compiler/parser/RdrHsSyn.hs

```diff
@@ -164,6 +165,18 @@ import Module
 
 #include "HsVersions.h"
 
+isDamlGenerated :: HasOccName a => a -> Bool
+isDamlGenerated namedThing =
```

Comment:
I'm wondering now if we could instead use a single prefix for all daml-generated stuff, say, `$daml$` for `$daml$_method_[...]`, `$daml$_choice_[...]`, etc. I don't want to shed too many bikes on this though.